test/custom-task-ctrls/wait-task-beta/cmd/controller/main.go

```diff
@@ -46,9 +46,34 @@ func newController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 		}
 	})
 
+	apiVersion := "wait.testing.tekton.dev/v1beta1"
+	kind := "Wait"
+
 	customruninformer.Get(ctx).Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: tkncontroller.FilterRunRef("wait.testing.tekton.dev/v1beta1", "Wait"),
-		Handler:    controller.HandleAll(impl.Enqueue),
+		// TODO: Replace with the following once tkncontroller.FilterCustomRunRef is merged (fromhttps://github.com/tektoncd/pipeline/pull/5822)
+		//      and released.
+		// FilterFunc: tkncontroller.FilterCustomRunRef("wait.testing.tekton.dev/v1beta1", "Wait"),
+		FilterFunc: func(obj interface{}) bool {
+			r, ok := obj.(*v1beta1.CustomRun)
+			if !ok {
+				// Somehow got informed of a non-Run object.
+				// Ignore.
+				return false
+			}
+			if r == nil || (r.Spec.CustomRef == nil && r.Spec.CustomSpec == nil) {
```

Comment:
Wanna confirm if my understanding is correct:

End users are allowed to define a CustomRun without `CustomRef` and `CustomSpec` specified right? For example: https://github.com/tektoncd/experimental/blob/main/wait-task/example-run.yaml